File: app/src/main/java/com/group15/fitnesstracker/db/dao/TrainsDao.java
package com.group15.fitnesstracker.db.dao;

import com.group15.fitnesstracker.db.Trains;

import java.util.List;

import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;
import io.reactivex.Completable;
import io.reactivex.Maybe;

public interface TrainsDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public Completable insertTrains(Trains... trains);

    @Update
    public Completable updateTrains(Trains... trains);

    @Delete
    public Completable deleteTrains(Trains... trains);

    @Query("SELECT * FROM Trains")
    public Maybe<List<Trains>> loadAllTrains();
}

File: app/src/main/java/com/group15/fitnesstracker/db/dao/TrainsDao.kt
package com.group15.fitnesstracker.db.dao

import androidx.room.*
import com.group15.fitnesstracker.db.Trains

import io.reactivex.Completable
import io.reactivex.Maybe

@Dao
interface TrainsDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertTrains(vararg trains: Trains): Completable

    @Insert
    fun addUsersToTrain(vararg train: Trains): Completable

    @Update
    fun updateTrains(vararg trains: Trains): Completable

    @Delete
    fun deleteTrains(vararg trains: Trains): Completable

    @Query("SELECT * FROM Trains")
    fun loadAllTrains(): Maybe<List<Trains>>
}
